airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -117,9 +106,10 @@ class GoogleAnalyticsDataApiAbstractStream(HttpStream, ABC):
     url_base = "https://analyticsdata.googleapis.com/v1beta/"
     http_method = "POST"
 
-    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, *, authenticator: Union[requests.auth.AuthBase, auth.HttpAuthenticator], config: Mapping[str, Any], **kwargs):
+        super().__init__(authenticator, **kwargs)
         self._config = config
+        self.saved_authenticator = authenticator
```

Comment:
what is the reason having this in  the code, it's not used elsewhere?

Reply:
If you need to downstream the `authenticator` please make it the part of the  `config` and call it from `self._config`.

Reply:
https://github.com/airbytehq/airbyte/blob/9ed7ddf6d02bd1bccbc5485363855312be65ec49/airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py#L95